### ISSUE TITLE
`app_configs` additional clickable roles is not respected

### ISSUE THREAD

**Issue by y3owk1n** [y3owk1n/neru#690]
### Pre-flight checks

- [x] I have searched existing issues and this bug has not been reported yet

### Neru version

main

### Operating system

macOS

### OS version

macOS 26.4

### Navigation mode

Hints

### What happened?

Setting the following, `AXTabGroup` does not get populated to the clickable roles properly. This is a bug introduced after a massive refactoring for linux support.

```toml
[[hints.app_configs]]
bundle_id = "net.imput.helium"
additional_clickable_roles = ["AXTabGroup"]
ignore_clickable_check = false

[hints.app_configs.hotkeys]
"Return" = ["action left_click", "exec sleep 3.5", "hints"]
```

### What did you expect?

`AXTabGroup` should be clickable.

### Steps to reproduce

1. Launch Neru
2. Set the config below
3. Go to Helium browser, create a tab group and launch hint

### Config (relevant sections)

```toml
[[hints.app_configs]]
bundle_id = "net.imput.helium"
additional_clickable_roles = ["AXTabGroup"]
```
